### DISCUSSION TITLE
Query as Prefix

### DISCUSSION THREAD

**Discussion by Serverfrog** [YOURLS/YOURLS#2887] [Q&A] [answered]
Is it Possible to have a Query as Prefix before generated shortlink?

What i try to do:
I want to have https://example.com/watch?XXXXX where `watch?` is just a prefix and `XXXXX` are randomly generated Base62 String with a length (should be more than 4, but not 50, if you know what i mean )

Is this with YOURLS possible?

I could think a Workaround with the prefix via .htaccess but im not sure if that is possible

**Comment by ozh** [comment, marked as answer]
I don't understand your question, but the answer is : yes. See https://github.com/YOURLS/YOURLS/wiki/Possible-With-A-Plugin

**Reply by Serverfrog** [reply]
I try to rephrase the question:

I wanted to Make/Use the Plugin capability to have a prefix for the ID of the URL that got shortend with a randomly generated Base62 String.

Both of these things are nearly possible with two plugins. One which prefixes and Another who Randomized the ID's.
But the prefix make it so that the ID will be the Query Parameter instead of an URL Path, and i could not find anything that could help me there to use the Query Parameter as ID. No Filter or Action i (shortly tried) to use worked and it seems it is multiple problems there i can not really tackle or know. Mostly because its a bit of a hassle as a documention to just have

"This is Action XY and it is used at Line Z" is not helping that much, without try to read the entire project....

**Reply by ozh** [reply]
Well, your first question was "is it possible with YOURLS", and the answer was and still is : "Yes, it is possible. With a plugin" 

Now you "rephrased" the question : it's not even a question anymore. What is your question? 😕

**Reply by Serverfrog** [reply]
Well, the Answer was "Yhea, as a Programmer you can do things" which was not so helpfull. Especially because i don't know if its really Possible as a Plugin or do i need to [Modify the Core Foles](https://github.com/YOURLS/YOURLS/wiki/Dont-Hack-Core).

Ok Rephrasing was maybe the wrong words. I tried to tell you what i try to achieve, and that i don't know where i could find information about the different kinds of Action and Filters and What they do when. That is because there exist only a list Of the Names with there LineNumber where they occure without any real information about them.

The Main Problem i got was that YOURLS doesn't care about Query paramters and just throws them away. There exists a Plugin which can append the Query Parameters that YOURLS got to the URL you are redirect but not much more (and it is not what im try to do).

So now to the simple Question i tried: can you please pinpoint me to a location (beside "Read the Sourcecode") where i could find good information what i could try to implement that feature.
Feature: using the Query as ID, instead of the Path

**Reply by ozh** [reply]
You definitely don't need to modify core files and virtually everything is possible with a plugin.

2 options to find a "location where you find good information" : 
- check others' plugins and how they work and code from there (particularly plugins playing with query strings)
- find a plugin coder to do it for you

I also respectfully suggest you put some more efforts into writing and proofreading your posts, which are quite hard to understand to be honest (typos, missing words and random Capitals).

**Reply by Serverfrog** [reply]
Ok.
Please understand that im not that used to write in English but thanks.

I will try to see, i think i will then use something different because it don't seem that a useful Documentation is existing beside Plugins of others as far as i understand you there.

I tried the "Discussion" section in the hope to have a more relaxed discussion where it isn't like everything is an attack on somebody but it seems it isn't

**Reply by ozh** [reply]
There really is [some documentation](https://github.com/YOURLS/YOURLS/wiki#for-developers) besides all the plugins themselves.

Sorry if you felt attacked, not my intention and I don't think my messages were written this way, but to each their own I guess :)

**Reply by Serverfrog** [reply]
Na its ok now. 
I was talking about exactly that documentation. To be precises about [the Hook List](https://yourls.org/hooklist.php) which is just a List of the existing Hooks to the source code lines which is not really a Documentation for this kind.

I got it myself easier and programmed that thing now [myself ](https://gitlab.com/Serverfrog/goshort)which should be the same speed as reading into that source code to find what Hook exactly i should use how it is then done.

Then the riding on how things of myself are written was also not the best kind of communication.
